### PR TITLE
feat: added code snippets extraction

### PR DESCRIPTION
**Pull request by Fagtoy**
Added possibility to extract code snippets from drafts into some directory in which will be created a new directory for every draft with its code snippets. Code snippets will not be extracted if ```src_id``` argument of ```YangModuleExtractor``` is a URL because it's not clear where to store code snippets of such a draft

### CODE REVIEW

**Comment by richardzilincikPantheon** [comment]
There doesn't seem to be a way to enable this from the command line tool. You should add the option to do so to `bin/xym`.

**Comment by Fagtoy** [comment]
> There doesn't seem to be a way to enable this from the command line tool. You should add the option to do so to `bin/xym`.

fixed

File: bin/xym [hunk 1]
```diff
@@ -22,8 +23,8 @@ def run():
     parser.add_argument(
         "--strict", type=bool, default=False,
         help='Optional flag that determines syntax enforcement; '
-        "'If set to 'True', the <CODE BEGINS> / <CODE ENDS> "
-        "tags are required; default is 'False'")
+             "'If set to 'True', the <CODE BEGINS> / <CODE ENDS> "
```
Comment (einarnn) [inline bin/xym:26]: In general, please try not to make unnecessary whitespace changes. They add to the noise when reviewing PRs.

File: xym/xym.py [hunk 2]
```diff
@@ -17,20 +17,31 @@
 
 from .yangParser import create_context
 
-__author__    = 'jmedved@cisco.com, calle@tail-f.com, bclaise@cisco.com, einarnn@gmail.com'
+__author__ = 'jmedved@cisco.com, calle@tail-f.com, bclaise@cisco.com, einarnn@gmail.com'
```
Comment (einarnn) [inline xym/xym.py:20]: Again, please avoid unnecessary whitespace changes.

**Review by einarnn — APPROVED** [review]
This PR took a lot longer for me to review than should have been necessary due to a large number of seemingly unnecessary whitespace changes and other cosmetic changes that seem to be based on the preference of the code editor. While I appreciate that some of these changes may improve the look of the code, it would be appreciated -- for ease of PR review -- if this approach is not taken for subsequent PRs.
